lib/src/model.dart

```diff
@@ -4054,6 +4056,109 @@ abstract class ModelElement extends Canonicalization
     }
   }
 
+  /// Replace &#123;@youtube ...&#125; in API comments with some HTML to embed
+  /// a YouTube video.
+  ///
+  /// Syntax:
+  ///
+  ///     &#123;@youtube WIDTH HEIGHT URL&#125;
+  ///
+  /// Example:
+  ///
+  ///     &#123;@youtube 560 315 https://www.youtube.com/watch?v=oHg5SJYRHA0&#125;
+  ///
+  /// Which will embed a YouTube player into the page that plays the specified
+  /// videos.
+  ///
+  /// The width and height must be integers specifying the dimensions of the
+  /// video in pixels.
+  ///
+  /// The following YouTube URL formats are supported:
+  ///
+  ///  * https://www.youtube.com/watch?v=oHg5SJYRHA0 (as shown in the browser's
+  ///    address bar)
+  ///  * https://youtu.be/oHg5SJYRHA0 (as shown in the "Share" dialog)
+  ///  * https://www.youtube.com/embed/oHg5SJYRHA0 (as shown in the "Share ->
+  ///    Embed" dialog)
+  String _injectYouTube(String rawDocs) {
+    // Matches all youtube directives (even some invalid ones). This is so
+    // we can give good error messages if the directive is malformed, instead of
+    // just silently emitting it as-is.
+    final RegExp basicAnimationRegExp = new RegExp(r'''{@youtube\s+([^}]+)}''');
+
+    // Matches YouTube IDs from supported YouTube URLs.
+    final RegExp validYouTubeUrlRegExp = new RegExp(
+        'https://((youtu\.be/)|(www\.youtube\.com/((watch\\?v=)|(embed/))))(.+)');
+
+    return rawDocs.replaceAllMapped(basicAnimationRegExp, (basicMatch) {
+      final ArgParser parser = new ArgParser();
+      final ArgResults args = _parseArgs(basicMatch[1], parser, 'youtube');
+      if (args == null) {
+        // Already warned about an invalid parameter if this happens.
+        return '';
+      }
+      final List<String> positionalArgs = args.rest.sublist(0);
+      if (positionalArgs.length != 3) {
+        warn(PackageWarning.invalidParameter,
+            message: 'Invalid @youtube directive, "${basicMatch[0]}"\n'
+                'YouTube directives must be of the form "{@youtube WIDTH '
+                'HEIGHT URL}"');
+        return '';
+      }
+
+      int width;
+      try {
+        width = int.parse(positionalArgs[0]);
```

Comment:
Maybe check width and height to make sure they're both greater than zero, and under 10,000 or so?

Reply:
Added a check for greater than zero. Imposing an upper limit seems really arbitrary, so I omitted that one.